enhancements/ocp-coreos-layering.md

```diff
@@ -0,0 +1,485 @@
+- [ ] Avoid breaking existing workflow via MachineConfig (including extensions)
+- [ ] Avoid overwriting existing custom modifications (such as files managed by other operators) during upgrades
+
+### Non-Goals
+
+- While the base CoreOS layer/ostree-container tools will be usable outside of OpenShift, this enhancement does not cover or propose any in-cluster functionality for exporting the forward-generated image outside of an OpenShift cluster. In other words, it is not intended to be booted (`$ rpm-ostree rebase <image>`) from outside of a cluster.
+- This proposal does not cover generating updated "bootimages"; see https://github.com/openshift/enhancements/pull/201
+- Don't change existing workflow for RHEL worker nodes
+
+## Proposal
+
+**NOTE: Nothing in this proposal should be viewed as final.  It is highly likely that details will change.  It is quite possible that larger architectural changes will be made as well.**
+
+1. The `machine-os-content` shipped as part of the release payload will change format to the new "native ostree-container" format, in which the OS content appears as any other OCI/Docker container.  
+  (In contrast today, the existing `machine-os-content` has an ostree repository inside a UBI image, which is hard to inspect and cannot be used for derived builds).  For more information, see [ostree-rs-ext](https://github.com/ostreedev/ostree-rs-ext/) and [CoreOS layering](https://github.com/coreos/enhancements/pull/7). 
+2. Documentation and tooling will be available for generating derived images from this base image
+3. This tooling will be used by the MCO to create derived images (more on this in a separate enhancement)
+
+
+### User Stories
+
+#### What works now continues to work
+
+An OpenShift administrator at example.corp is happily using OpenShift 4 (with RHEL CoreOS) in several AWS clusters today, and has only a small custom MachineConfig object to tweak host level auditing.  They do not plan to use any complex derived builds, and expect that upgrading their existing cluster continues to work and respect their small audit configuration change with no workflow changes.
+
+#### Adding a 3rd party security scanner/IDS
+
+example.bank's security team requires a 3rd party security agent to be installed on bare metal machines in their datacenter.  The 3rd party agent comes as an RPM today, and requires its own custom configuration.  While the 3rd party vendor has support for execution as a privileged daemonset on their roadmap, it is not going to appear soon. 
+
+After initial cluster provisioning is complete, the administrators at example.bank supply a configuration that adds a repo file to `/etc/yum.repos.d/agentvendor.repo` and requests installation of a package named `some-3rdparty-security-agent` as part of a container build.
+
+The build is successful, and is included in both the control plane (master) and worker pools, and is rolled out in the same way the MCO performs configuration and OS updates today.
```

Comment:
i'm wondering how this will be supported in the context of heterogeneous clusters as the internal image registry will not support manifestlisted images and builds doesn't support it either: https://github.com/openshift/enhancements/pull/1014/files#r797900391

Reply:
Pushed a blurb on this